libraries/src/MVC/Model/AdminModel.php

```diff
@@ -1443,6 +1445,15 @@ public function save($data)
             }
         }
 
+        if ($this instanceof VersionableModelInterface) {
+            // Merge table data and data so that we write all data to the history
+            $tableData = ArrayHelper::fromObject($table);
+
+            $historyData = array_merge($tableData, $data);
```

Comment:
```suggestion
            $historyData = array_merge($tableData, $data);
            
            // We have to set the key for new items, would be always 0 otherwise
            $historyData[$key] = $this->getState($this->getName() . '.id');

```

This should fix the problem with storing the first version and the disappearing buttons.

Reply:
didnt appear to make any changes for me

Reply:
Could you please try with a completely new entry. 
This can not fix a testing entry, that was saved with the 0. But if you try with a new one this should not be the case anymore.

Reply:
yes it does work with a brand new article

Reply:
so that would mean we need to update all existing content on a site which is never a good idea

Reply:
Yes you are absolutely right, that would be not a good idea. 
Does the problem occur ‘only’ with entries created with a previous version of this PR, or also with entries that were already versioned before this PR was patched?

Reply:
it might be ok
I created 2 articles before this PR - one with a version history and one without
I applied the pr and this fix and it would appear that I could restore to the original version without problem

BUT please verify this yourself

Reply:
Thank you for double checking.
I tested on a new testing site setup (without PR). Installed sample data blog, edited the typography article a view times and tested restore of version.
Installed the PR and added the fix - doing the same as above.
All works fine for me - except, of course, for the compare and preview view, which has not yet been revised.

@heelc29 If you have time, could you also check again if this solves the button and first version restore issue?

Reply:
Yes, this is now working - but still other issues